### PR TITLE
Implement hotkey-toggle Option with statuses

### PR DESCRIPTION
**Pull request by Stoneguard001**
This code adds an option to toggle hotkeys instead of having to hold them down. It also provides a rollover list of hot-keyed options that show their status and can be clicked to turn on or off. 

This code also contains a simple sort for the hotkey options and pandaniel's reordered list of options in settings.js. If I should remove those, please let me know.

### CODE REVIEW

**Comment by Aeristoka** [comment]
@Stoneguard001 Could you rebase this PR? Quite a few things have changed since you submitted it.

**Comment by Stoneguard001** [comment]
I will see what I can do. I just assumed it was no longer desired or no one had looked at it. It's been a while since I looked at the code, but will work on it!

**Comment by kakaroto** [comment]
> I will see what I can do. I just assumed it was no longer desired or no one had looked at it. It's been a while since I looked at the code, but will work on it!

nope, it's still desired, just been a crazy few months, I did look at it a couple of weeks ago, had a few small comments that I didn't publish as I was not done with the review and was going to fix it up locally anywhere to avoid a long back-and-forth cycle for minor edits.

File: src/dndbeyond/base/utils.js [hunk 1]
```diff
@@ -362,6 +362,50 @@ function updateRollTypeButtonClasses(character) {
...
+        someSet |= modifier;
+        if (!adn.includes(key)){
+            let ability = character_settings[key.replace('option-','')];
+            if (ability !== undefined) {
+                ability = ability.title;
+            } else {
+                ability = BINDING_NAMES[key] === undefined ? key : BINDING_NAMES[key];
+            }
+            if (ability.length > 32)
+                ability = ability.substring(0, 31) + "...";
```
[the remark this answers is not in the document]
Comment (kakaroto) [reply src/dndbeyond/base/utils.js:389]: Note: I wasn't able to do that properly because CSS is misbehaving obviously, so I left your solution for it.

**Review by kakaroto — APPROVED** [review]
It took so long but I finally went through this and did a review of the code. I didn't want to go through a whole back-and-forth on this, so I just fixed all the things I noticed, refactored stuff a little bit and pushed it at the same time, so while there's a bunch of comments here, you have nothing to do, otherwise than see what I would have done differently.
Thanks again for doing this, it looks great and for being patient with the review!

**Comment by Stoneguard001** [comment]
Sorry about the .gitignore not being ignored (sigh, TFS is my most familiar code management system). All the nomenclature changes make perfect sense; I am horrible at naming things. My original intent was to be able to keyclick to turn on something in the dice roll, and once the dice were rolled, it gets turned off. I'm not sure if the changes will still implement that, but as long as it works as you intend, that's fine. 

The only thing I vaguely remember is implementing the $(document).on and .off in character.js because otherwise there are dangling listeners out there when the browser is resized that cause things to fire twice. Of course, this was back in December, so that may be different now.

**Comment by kakaroto** [comment]
Yep, I kept that, we can still keyclick to turn things on. I made it also appear for when the user doesn't have sticky hotkeys enabled, so they can keyclick there to enable something. It doesn't remove it after a roll though in that case.

**Pull request closed without merging**